core/src/forwarding_stage.rs

```diff
@@ -203,29 +223,55 @@ impl ForwardingClient for ConnectionCacheClient {
 
 pub(crate) fn spawn_forwarding_stage(
     receiver: Receiver<(BankingPacketBatch, bool)>,
-    connection_cache: Arc<ConnectionCache>,
+    client: ForwardingClientOption<'_>,
     vote_client_udp_socket: UdpSocket,
     root_bank_cache: RootBankCache,
     forward_address_getter: ForwardAddressGetter,
     data_budget: DataBudget,
 ) -> JoinHandle<()> {
     let vote_client = VoteClient::new(vote_client_udp_socket, forward_address_getter.clone());
-
-    let non_vote_client = ConnectionCacheClient::new(connection_cache, forward_address_getter);
-    let forwarding_stage = ForwardingStage::new(
-        receiver,
-        vote_client,
-        non_vote_client.clone(),
-        root_bank_cache,
-        data_budget,
-    );
-    Builder::new()
-        .name("solFwdStage".to_string())
-        .spawn(move || forwarding_stage.run())
-        .unwrap()
+    match client {
+        ForwardingClientOption::ConnectionCache(connection_cache) => {
+            let non_vote_client =
+                ConnectionCacheClient::new(connection_cache, forward_address_getter);
+            let forwarding_stage = ForwardingStage::new(
+                receiver,
+                vote_client,
+                non_vote_client.clone(),
+                root_bank_cache,
+                data_budget,
+            );
+            Builder::new()
+                .name("solFwdStage".to_string())
+                .spawn(move || forwarding_stage.run())
+                .unwrap()
+        }
+        ForwardingClientOption::TpuClientNext((
+            stake_identity,
+            tpu_client_socket,
+            runtime_handle,
+        )) => {
+            let non_vote_client = TpuClientNextClient::new(
+                runtime_handle,
+                forward_address_getter,
+                Some(stake_identity),
+                tpu_client_socket,
+            );
+            let forwarding_stage = ForwardingStage::new(
+                receiver,
+                vote_client,
+                non_vote_client.clone(),
```

Comment:
Do we need to clone here? It looks like this is the only use of the non_vote_client. If we don't need to, can we also remove the Clone implementation?

Reply:
You are right